test/src/test/java/org/corfudb/integration/LogReplicationIT.java

```diff
@@ -383,8 +392,92 @@ private void verifyNoData(Map<String, Table<StringKey, IntValue, Metadata>> tabl
+        LogReplicationMetadata.SyncStatus syncStatus = null;
+        Table<LogReplicationMetadata.ReplicationStatusKey, LogReplicationMetadata.ReplicationStatusVal, Message> replicationStatusTable;
+        replicationStatusTable = srcCorfuStore.openTable(CORFU_SYSTEM_NAMESPACE,
+                REPLICATION_STATUS_TABLE,
+                LogReplicationMetadata.ReplicationStatusKey.class,
+                LogReplicationMetadata.ReplicationStatusVal.class,
+                null,
+                TableOptions.fromProtoSchema(LogReplicationMetadata.ReplicationStatusVal.class));
+
+        try (TxnContext txn = srcCorfuStore.txn(CORFU_SYSTEM_NAMESPACE)) {
+            CorfuStoreEntry<LogReplicationMetadata.ReplicationStatusKey, LogReplicationMetadata.ReplicationStatusVal, Message>
+                    entry = txn.getRecord(replicationStatusTable, key);
+            if (entry.getPayload() != null) {
+                syncStatus = entry.getPayload().getStatus();
+            }
+            txn.commit();
+        }
+
+        Assert.assertEquals(expectedStatus, syncStatus);
+    }
+
     /* ***************************** LOG REPLICATION IT TESTS ***************************** */
 
+    /**
+     * This test helps to emulate the raceCondition where statusSyncUpdater thread gets precedence over the state transition.
+     * A statusSyncUpdater service is run independently without the FSM and
+     * NOT_STARTED status is verified from the replicationStatus table.
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testSyncStatusBeforeSyncState() throws Exception {
+        setupEnv();
+
+        LogReplicationMetadataManager srcManager = new LogReplicationMetadataManager(srcTestRuntime, 0, ACTIVE_CLUSTER_ID);
+        LogReplicationConfigManager tableManagerPlugin = new LogReplicationConfigManager(srcTestRuntime);
+        LogReplicationConfig config = new LogReplicationConfig(tableManagerPlugin, BATCH_SIZE,
+                SMALL_MSG_SIZE, MAX_CACHE_NUM_ENTRIES);
+
+        LogEntryReader logEntryReader = new TestLogEntryReader();
+        LogReplicationAckReader ackReader = new LogReplicationAckReader(srcManager, config, srcDataRuntime, REMOTE_CLUSTER_ID);
+        ackReader.setLogEntryReader(logEntryReader);
+
+        try {
+            ackReader.setSyncType(LogReplicationMetadata.ReplicationStatusVal.SyncType.SNAPSHOT);
+            ackReader.startSyncStatusUpdatePeriodicTask();
+            TimeUnit.SECONDS.sleep(WAIT_INTERVAL.getSeconds());
+        } catch (InterruptedException e) {
+            System.out.println("Sleep interrupted: " + e);
+        } finally {
+            ackReader.shutdown();
+        }
+
+        verifyPreviousSnapshotStatus(LogReplicationMetadata.SyncStatus.NOT_STARTED);
+        verifyCurrentSyncStatus(LogReplicationMetadata.SyncStatus.NOT_STARTED);
+
+        cleanEnv();
+    }
+
```

Comment:
Another test can also be when the FSM state is `INITIALIZED`, but statusSyncUpdater has started.
You can include this bit in one of the exisiting tests in this suite.
Like in `startSnapshotSync()` after creation of SourceManager but before replication has started.

Reply:
As we discussed yesterday, it would be the same flow there as well. Adding the same checks for snapshot sync just before executing a snapshot sync would add an overload for most of the other ITs.